### DISCUSSION TITLE
Allow customizing the S3 Backend for a BudibaseDB in builder

### DISCUSSION THREAD

**Discussion by daawaan4x** [Budibase/budibase#5137] [Ideas]
**Describe the feature request**

Since the builder already allows adding an S3 storage service as a datasource, I think MinIO should be decoupled from BudibaseDB and it should be configurable in the builder to replace MinIO with another S3 storage service (e.g. replacing MinIO with Digital Ocean Spaces for Digital Ocean Deployment #5121)

**Comment by mike12345567** [comment]
Hi @daawaan - this is something that we allow in self hosted environments, you can change it already using the combination of `AWS_REGION`, `MINIO_URL`, `MINIO_ACCESS_KEY` and `MINIO_SECRET_KEY` on the `budibase/apps` instance in your environment. You can see the full list of supported environment variables here: https://github.com/Budibase/budibase/blob/develop/packages/server/src/environment.js

For the Cloud, this could be interesting, but I don't see this change being made anytime soon as we currently utilise S3 for this already - there could be some difficulties around being able to support multiple object stores at the same time.

**Comment by mike12345567** [comment]
Moved to ideas discussion so others can also discuss.

**Reply by liongkj** [reply]
do you have a walk through / documentation that i can refer to?
